video/out/drm_common.c

```diff
@@ -1240,6 +1365,55 @@ double vo_drm_get_display_fps(struct vo_drm_state *drm)
     return mode_get_Hz(&drm->mode.mode);
 }
 
+bool vo_drm_set_hdr_metadata(struct vo *vo)
+{
+    struct vo_drm_state *drm = vo->drm;
+    struct mp_image_params target_params = vo_get_target_params(vo);
+    if (!vo->target_params || pl_color_space_equal(&target_params.color, &drm->target_params.color))
+        return false;
+
+    destroy_hdr_blob(drm);
+    drm->target_params = target_params;
+    if (!target_params_supported_by_display(drm)) {
+        restore_sdr(drm);
+        return false;
+    }
+
+
+    struct drm_atomic_context *atomic_ctx = drm->atomic_context;
```

Comment:
Shouldn't this be checked first?

Reply:
It doesn't need to be checked at all actually. We always have an atomic context here.